app/controllers/users_controller.rb

```diff
@@ -11,22 +10,68 @@ def index
     @users = User.all
   end
 
+  def new
+    if current_user.admin?
+      @user = User.new
+    else
+      set_flash_message :alert, :must_be_admin
```

Comment:
This is a nice candidate to encapsulate in some sort of helper, or method in ApplicationController, if you use it more than once.